va/validation-authority.go

```diff
@@ -289,7 +288,7 @@ func (va *ValidationAuthorityImpl) fetchHTTP(identifier core.AcmeIdentifier, pat
 		}
 		return emptyBody, challenge, err
 	}
-
+	httpResponse.Body.Close()
```

Comment:
It might be better to defer this before calling `ioutil.ReadFile` so it'd also get called after a read error?